include/nbl/video/IAPIConnection.h

```diff
@@ -12,13 +14,14 @@ namespace video
 class IAPIConnection : public core::IReferenceCounted
 {
 public:
-    // TODO implement in some source file in src/nbl/...
     static core::smart_refctd_ptr<IAPIConnection> create(E_API_TYPE apiType, uint32_t appVer, const char* appName);
 
     virtual E_API_TYPE getAPIType() const = 0;
 
     virtual core::SRange<const core::smart_refctd_ptr<IPhysicalDevice>> getPhysicalDevices() const = 0;
 
+    virtual core::smart_refctd_ptr<ISurface> createSurface(system::IWindow* window) const = 0;
```

Comment:
wrong way to go about it?

Reply:
?

Reply:
window or surface itself should create the surface... again because of multiple backend windows etc.

Reply:
multiple backend windows? surfaces are created by api connections exactly because of multiple backends. And regarding to windows - it always boils down to os-specific handle so that's why we have IWindowWin32 for Windows platform for example with gwtNativeHandle () api

Reply:
then how does the user create their own surfaces from a Qt/SDL/etc. window without modifying API connection code?

Reply:
by deriving from IWindowWin32

Reply:
but an SDL/Qt Window will not give the native handle to Nabla, just spoof a surface and swapchain (isolation of contexts/devices)

Reply:
TODO?